InnerEye/ML/utils/checkpoint_handling.py

```diff
@@ -228,3 +238,179 @@ def get_local_checkpoints_path_or_download(self) -> List[Path]:
             if not checkpoint_path or not checkpoint_path.is_file():
                 raise FileNotFoundError(f"Could not find the weights file at {checkpoint_path}")
         return checkpoint_paths
+
+
+def download_checkpoints_to_temp_folder(run: Optional[Run] = None, workspace: Optional[Workspace] = None) -> Path:
+    """
+    Downloads all files with the outputs/checkpoints prefix of the given run to a temporary folder.
+    In distributed training, the download only happens once per node.
+
+    :param run: If provided, download the files from that run. If omitted, download the files from the current run
+        (taken from RUN_CONTEXT)
+    :param workspace: The AML workspace where the run is located. If omitted, the hi-ml defaults of finding a workspace
+        are used (current workspace when running in AzureML, otherwise expecting a config.json file)
+    :return: The path to which the files were downloaded. The files are located in that folder, without any further
+        subfolders.
+    """
+    run = run or RUN_CONTEXT
+    # Downloads should go to a temporary folder because downloading the files to the checkpoint folder might
+    # cause artifact conflicts later.
+    temp_folder = Path(tempfile.mkdtemp())
+    checkpoint_prefix = f"{DEFAULT_AML_UPLOAD_DIR}/{CHECKPOINT_FOLDER}/"
```

Comment:
I wonder if we should make this an argument of the function, with this value as the default, so that it is possible to specify a custom checkpoint path?